tla-bmcmt/src/main/scala/at/forsyte/apalache/tla/bmcmt/ModelChecker.scala

```diff
@@ -506,22 +510,38 @@ class ModelChecker(typeFinder: TypeFinder[CellT],
     }
   }
 
-  private def dumpCounterexample(): String = {
-    val filename = "counterexample.txt"
+  private def dumpCounterexample(notInv: TlaEx): String = {
+    // Assumption: stack cannot be empty
+    val filename = if(tlc) "MC.out" else "counterexample.tla"
     val writer = new PrintWriter(new FileWriter(filename, false))
+
+    val nextStates = new ListBuffer[NextState]()
     for (((state, oracle), i) <- stack.reverse.zipWithIndex) {
       val decoder = new SymbStateDecoder(solverContext, rewriter)
       val transition = oracle.evalPosition(solverContext, state)
-      writer.println(s"State $i (from transition $transition):")
-      writer.println("--------")
       val binding = decoder.decodeStateVariables(state)
-      val args = binding.toList.sortBy(_._1).flatMap(p => List(ValEx(TlaStr(p._1)), p._2))
-      if (args.nonEmpty) {
-        val record = OperEx(TlaFunOper.enum, args :_*)
-        new PrettyWriter(writer).write(record)
-      }
-      writer.println("\n========\n")
+      nextStates += ((transition.toString, binding))
+    }
+    if(tlc) {
+      writer.println(
+        s"""@!@!@STARTMSG 2262:0 @!@!@
+          |Created by Apalache on ${Calendar.getInstance().getTime}
+          |@!@!@ENDMSG 2262 @!@!@""".stripMargin
+      )
```

Comment:
I believe that this header should be implemented in TlcCounterexampleWriter. Perhaps, as a separate method, if you like